x-pack/plugins/security_solution/server/lib/timeline/saved_object/timelines/index.ts

```diff
@@ -307,74 +307,61 @@ export const persistFavorite = async (
-      timeline = savedTimeline;
-    }
+  let timeline: SavedTimeline = {};
+  if (timelineId != null) {
+    const {
+      eventIdToNoteIds,
+      notes,
+      noteIds,
+      pinnedEventIds,
+      pinnedEventsSaveObject,
+      savedObjectId,
+      version,
+      ...savedTimeline
+    } = await getBasicSavedTimeline(request, timelineId);
+    timelineId = savedObjectId; // eslint-disable-line no-param-reassign
+    timeline = savedTimeline;
+  }
 
-    const userFavoriteTimeline = {
-      keySearch: userName != null ? convertStringToBase64(userName) : null,
-      favoriteDate: new Date().valueOf(),
-      fullName,
-      userName,
-    };
-    if (timeline.favorite != null) {
-      const alreadyExistsTimelineFavoriteByUser = timeline.favorite.findIndex(
-        (user) => user.userName === userName
-      );
-
-      timeline.favorite =
-        alreadyExistsTimelineFavoriteByUser > -1
-          ? [
-              ...timeline.favorite.slice(0, alreadyExistsTimelineFavoriteByUser),
-              ...timeline.favorite.slice(alreadyExistsTimelineFavoriteByUser + 1),
-            ]
-          : [...timeline.favorite, userFavoriteTimeline];
-    } else if (timeline.favorite == null) {
-      timeline.favorite = [userFavoriteTimeline];
-    }
+  const userFavoriteTimeline = {
+    keySearch: userName != null ? convertStringToBase64(userName) : null,
+    favoriteDate: new Date().valueOf(),
+    fullName,
+    userName,
+  };
+  if (timeline.favorite != null) {
+    const alreadyExistsTimelineFavoriteByUser = timeline.favorite.findIndex(
+      (user) => user.userName === userName
+    );
 
-    const persistResponse = await persistTimeline(request, timelineId, null, {
-      ...timeline,
-      templateTimelineId,
-      templateTimelineVersion,
-      timelineType,
-    });
-    return {
-      savedObjectId: persistResponse.timeline.savedObjectId,
-      version: persistResponse.timeline.version,
-      favorite:
-        persistResponse.timeline.favorite != null
-          ? persistResponse.timeline.favorite.filter((fav) => fav.userName === userName)
-          : [],
-      templateTimelineId,
-      templateTimelineVersion,
-      timelineType,
-    };
-  } catch (err) {
-    if (getOr(null, 'output.statusCode', err) === 403) {
-      return {
-        savedObjectId: '',
-        version: '',
-        favorite: [],
-        code: 403,
-        message: err.message,
-      };
-    }
-    throw err;
+    timeline.favorite =
+      alreadyExistsTimelineFavoriteByUser > -1
+        ? [
+            ...timeline.favorite.slice(0, alreadyExistsTimelineFavoriteByUser),
+            ...timeline.favorite.slice(alreadyExistsTimelineFavoriteByUser + 1),
+          ]
+        : [...timeline.favorite, userFavoriteTimeline];
+  } else if (timeline.favorite == null) {
+    timeline.favorite = [userFavoriteTimeline];
   }
+
+  const persistResponse = await persistTimeline(request, timelineId, null, {
```

Comment:
Why remove the error handling here?  See this comment: https://github.com/elastic/kibana/pull/200633/files#r1851221603